templates/firstboot/mlnx_fw_update.py

```diff
@@ -16,25 +16,28 @@
 FW_VERSION_REGEX = r'FW Version:\s*\t*(?P<fw_ver>\d+\.\d+\.\d+)'
 RUNNING_FW_VERSION_REGEX = r'FW Version\(Running\):\s*\t*(?P<fw_ver>\d+\.\d+\.\d+)'
 PSID_REGEX = r'PSID:\s*\t*(?P<psid>\w+)'
+ARRAY_VALUE_REGEX = r'Array\[(?P<first_index>\d+)\.\.(?P<last_index>\d+)\]'
```

Comment:
not for this PR. but in the future we should create the Yaml file automatically from this python script using jinja to render from template. Then you should maintain the script in the yaml file. Again this is not for now, but for future enchantment